src/server/transaction.cc

```diff
@@ -1412,7 +1413,8 @@ void Transaction::CancelBlocking(std::function<OpStatus(ArgSlice)> status_cb) {
     return;
 
   coordinator_state_ |= COORD_CANCELLED;
-  local_result_ = status;
+  // don't use local_result_ because it can be overwirtten if we cancel ahead
+  block_cancel_result_ = status;
```

Comment:
seems you do not need `block_cancel_result_` at all. if you reached this point, `block_cancel_result_` will be set CANCELLED, is not it?